curator-framework/src/main/java/org/apache/curator/framework/imps/DelegatingCuratorFramework.java

```diff
@@ -0,0 +1,213 @@
+        return client.getZookeeperClient();
+    }
+
+    @Override
+    public ZookeeperCompatibility getZookeeperCompatibility() {
+        return client.getZookeeperCompatibility();
+    }
+
+    @Override
+    @Deprecated
+    public void clearWatcherReferences(Watcher watcher) {
+        client.clearWatcherReferences(watcher);
+    }
+
+    @Override
+    public boolean blockUntilConnected(int maxWaitTime, TimeUnit units) throws InterruptedException {
+        return client.blockUntilConnected(maxWaitTime, units);
+    }
+
+    @Override
+    public void blockUntilConnected() throws InterruptedException {
+        client.blockUntilConnected();
+    }
+
+    @Override
+    public ConnectionStateErrorPolicy getConnectionStateErrorPolicy() {
+        return client.getConnectionStateErrorPolicy();
+    }
+
+    @Override
+    public QuorumVerifier getCurrentConfig() {
+        return client.getCurrentConfig();
+    }
+
+    @Override
+    public SchemaSet getSchemaSet() {
+        return client.getSchemaSet();
+    }
+
+    @Override
+    public CompletableFuture<Void> postSafeNotify(Object monitorHolder) {
+        return client.postSafeNotify(monitorHolder);
+    }
+
+    @Override
+    public CompletableFuture<Void> runSafe(Runnable runnable) {
+        return client.runSafe(runnable);
+    }
+
+    @Override
+    NamespaceImpl getNamespaceImpl() {
+        return client.getNamespaceImpl();
+    }
+
+    @Override
+    void validateConnection(Watcher.Event.KeeperState state) {
+        client.validateConnection(state);
+    }
+
+    @Override
+    WatcherRemovalManager getWatcherRemovalManager() {
+        return client.getWatcherRemovalManager();
+    }
+
+    @Override
+    FailedDeleteManager getFailedDeleteManager() {
+        return client.getFailedDeleteManager();
+    }
+
+    @Override
+    FailedRemoveWatchManager getFailedRemoveWatcherManager() {
+        return client.getFailedRemoveWatcherManager();
+    }
+
+    @Override
+    public void logError(String reason, Throwable e) {
+        client.logError(reason, e);
+    }
+
+    @Override
+    byte[] getDefaultData() {
+        return client.getDefaultData();
+    }
+
+    @Override
+    CompressionProvider getCompressionProvider() {
+        return client.getCompressionProvider();
+    }
+
+    @Override
+    public boolean compressionEnabled() {
+        return client.compressionEnabled();
+    }
+
+    @Override
+    ACLProvider getAclProvider() {
+        return client.getAclProvider();
+    }
+
+    @Override
+    boolean useContainerParentsIfAvailable() {
+        return client.useContainerParentsIfAvailable();
+    }
+
+    @Override
+    EnsembleTracker getEnsembleTracker() {
+        return client.getEnsembleTracker();
+    }
+
+    @Override
+    NamespaceFacadeCache getNamespaceFacadeCache() {
+        return client.getNamespaceFacadeCache();
+    }
+
+    @Override
+    public <DATA_TYPE> void processBackgroundOperation(
```

Comment:
```suggestion
    <DATA_TYPE> void processBackgroundOperation(
```

.. and may need a re-format